ext/pdo_pgsql/pgsql_statement.c

```diff
@@ -513,6 +517,18 @@ static int pgsql_stmt_get_col(pdo_stmt_t *stmt, int colno, zval *result, enum pd
 #endif
 				ZVAL_LONG(result, ZEND_ATOL(ptr));
 				break;
+			case FLOAT4OID:
+			case FLOAT8OID:
+                if (len == sizeof("Infinity") - 1 && strncmp(ptr, "Infinity", len) == 0) {
+                    ZVAL_DOUBLE(result, ZEND_INFINITY);
+                } else if (len == sizeof("-Infinity") - 1 && strncmp(ptr, "-Infinity", len) == 0) {
+                    ZVAL_DOUBLE(result, -ZEND_INFINITY);
+                } else if (len == sizeof("NaN") - 1 && strncmp(ptr, "NaN", len) == 0) {
+                    ZVAL_DOUBLE(result, ZEND_NAN);
+                } else {
+                    ZVAL_DOUBLE(result, zend_strtod(ptr, NULL));
+                }
```

Comment:
```suggestion
                ZVAL_DOUBLE(result, zend_strtod(ptr, NULL));
```
isn't this enough? - https://github.com/php/php-src/blob/48f0b10f32ee52594685f9b885e0926fbccfa866/Zend/zend_strtod.c#L2726

Reply:
The float data type in PostgreSQL can return special values such as Infinity, -Infinity, and NaN. These special values need to be handled differently.

If no processing is done, the test results will be:

```log
TEST 1/1 [ext/pdo_pgsql/tests/float.phpt]
========DIFF========
--
         ["value8"]=>
         float(0.8)
         ["valuenan"]=>
009-     float(NAN)
009+     float(0)
         ["valueinfinity"]=>
011-     float(INF)
011+     float(0)
         ["valueninfinity"]=>
013-     float(-INF)
013+     float(0)
       }
     }
     array(7) {
--
========DONE========
```

Reply:
the special values should be handled by the `zend_strtod`

Reply:
It's a good idea, but I don't think it's appropriate to do it in this PR.

Reply:
correct me if i am wrong, but in https://github.com/php/php-src/blob/48f0b10f32ee5/Zend/zend_strtod.c#L2726 it is already handled

Reply:
It seems like Nan and Infinity don't always work in zend_strtod(), they are controlled by INFNAN_CHECK

https://github.com/php/php-src/blob/48f0b10f32ee5/Zend/zend_strtod.c#L2721

Reply:
I agree with @Yurunsoft here, however if zend_strtod is updated we can always update here later.